client/app/cloud/project/compute/infrastructure/virtualMachine/add/cloud-project-compute-infrastructure-virtualMachine-add.service.js

```diff
@@ -0,0 +1,181 @@
+class CloudProjectVirtualMachineAddService {
+    constructor ($q, $translate, CloudFlavorService, CloudImageService, ControllerModalHelper, OvhApiCloudProject, OvhApiCloudProjectInstance, OvhApiCloudProjectNetworkPrivateSubnet) {
+        this.$q = $q;
+        this.$translate = $translate;
+        this.CloudFlavorService = CloudFlavorService;
+        this.CloudImageService = CloudImageService;
+        this.ControllerModalHelper = ControllerModalHelper;
+        this.OvhApiCloudProject = OvhApiCloudProject;
+        this.OvhApiCloudProjectInstance = OvhApiCloudProjectInstance;
+        this.OvhApiCloudProjectNetworkPrivateSubnet = OvhApiCloudProjectNetworkPrivateSubnet;
+    }
+
+    getAugmentedImages (images) {
+        return _.map(_.uniq(images, "id"), this.CloudImageService.augmentImage);
+    }
+
+    getAugmentedFlavorsFilteredByType (flavors, type) {
+        return _.filter(_.map(_.filter(flavors, {
```

Comment:
Can you try to go from 2 filters to 1 filter?  If you struggle just forget this comment